### PR TITLE
mempool update to v2.5.0

### PR DESCRIPTION
**Pull request by openoms**
fixes: #4352

### CODE REVIEW

File: home.admin/config.scripts/bonus.mempool.sh [hunk 1]
```diff
@@ -135,40 +135,36 @@ if [ "$1" = "install" ]; then
...
 
   echo "# npm install for mempool explorer (frontend)"
 
-  cd frontend
-  sudo -u mempool NG_CLI_ANALYTICS=false npm install --no-optional
-  if ! [ $? -eq 0 ]; then
-      echo "FAIL - npm install did not run correctly, aborting"
-      exit 1
+  cd frontend || exit 1
+  if ! sudo -u mempool NG_CLI_ANALYTICS=false npm install --no-optional; then
```
Comment (cstenglein) [inline home.admin/config.scripts/bonus.mempool.sh]: Better use 
```npm ci```

File: home.admin/config.scripts/bonus.mempool.sh [hunk 2]
```diff
@@ -135,40 +135,36 @@ if [ "$1" = "install" ]; then
...
   echo "# npm install for mempool explorer (backend)"
 
-  cd ../backend/
-  sudo -u mempool NG_CLI_ANALYTICS=false npm install --no-optional
-  if ! [ $? -eq 0 ]; then
-      echo "# FAIL - npm install did not run correctly, aborting"
-      echo "result='failed npm install'"
-      exit 1
+  cd ../backend/ || exit 1
+  if ! sudo -u mempool NG_CLI_ANALYTICS=false npm install --no-optional; then
```
Comment (cstenglein) [inline home.admin/config.scripts/bonus.mempool.sh]: npm ci

File: home.admin/config.scripts/bonus.mempool.sh [hunk 3]
```diff
@@ -135,40 +135,36 @@ if [ "$1" = "install" ]; then
...
+    echo "# FAIL - npm install did not run correctly, aborting"
+    echo "result='failed npm install'"
+    exit 1
   fi
-  sudo -u mempool NG_CLI_ANALYTICS=false npm run build
-  if ! [ $? -eq 0 ]; then
-      echo "# FAIL - npm run build did not run correctly, aborting (2)"
-      echo "result='failed npm run build'"
-      exit 1
+  if ! sudo -u mempool NG_CLI_ANALYTICS=false npm run build; then
```
Comment (cstenglein) [inline home.admin/config.scripts/bonus.mempool.sh:164]: Instead of repeating the ng analytics command you could export it

export NG_CLI_ANALYTICS="false"

Comment (openoms) [reply home.admin/config.scripts/bonus.mempool.sh:164]: not sure of that works when running the command with a different user

File: home.admin/config.scripts/bonus.mempool.sh (hunk 1 above)
Comment (openoms) [reply home.admin/config.scripts/bonus.mempool.sh]: will test if that does make a difference with the failing angular build discussed in #4352

Comment (openoms) [reply home.admin/config.scripts/bonus.mempool.sh]: awesome that worked. Changed in https://github.com/raspiblitz/raspiblitz/pull/4353/commits/6232b6f41b73124717b6b9c6ada61e928080c90d

File: home.admin/config.scripts/bonus.mempool.sh [hunk 4]
```diff
@@ -422,73 +420,66 @@ fi
...
-          echo "FAIL - npm install did not run correctly, aborting"
-          exit 1
-      fi
-      sudo -u mempool NG_CLI_ANALYTICS=false npm run build
-      if ! [ $? -eq 0 ]; then
-          echo "FAIL - npm run build did not run correctly, aborting (4)"
-          exit 1
-      fi
+    cd /home/mempool/mempool/backend/ || exit 1
+    if ! sudo -u mempool NG_CLI_ANALYTICS=false npm install; then
```
Comment (cstenglein) [inline home.admin/config.scripts/bonus.mempool.sh]:
```suggestion
    if ! sudo -u mempool NG_CLI_ANALYTICS=false npm ci; then
```

File: home.admin/config.scripts/bonus.mempool.sh [hunk 5]
```diff
@@ -422,73 +420,66 @@ fi
...
+      echo "FAIL - npm run build did not run correctly, aborting (3)"
+      exit 1
+    fi
 
-      sudo mv /home/admin/mempool-config.json /home/mempool/mempool/backend/mempool-config.json
-      sudo chown mempool:mempool /home/mempool/mempool/backend/mempool-config.json
+    echo "# npm install for mempool explorer (frontend)"
 
+    cd ../frontend || exit 1
+    if ! sudo -u mempool NG_CLI_ANALYTICS=false npm install; then
```
Comment (cstenglein) [inline home.admin/config.scripts/bonus.mempool.sh]:
```suggestion
    if ! sudo -u mempool NG_CLI_ANALYTICS=false npm ci; then
```

**Review by cstenglein — CHANGES_REQUESTED** [review]

**Comment by cstenglein** [comment]
When installing mempool with that script, it works, but it doesn't add it to the menu & doesn't tick the box in the menu.

**Comment by openoms** [comment]
> When installing mempool with that script, it works, but it doesn't add it to the menu & doesn't tick the box in the menu.

can't reproduce this, works well now on amd64 and rpi5

**Comment by cstenglein** [comment]
> > When installing mempool with that script, it works, but it doesn't add it to the menu & doesn't tick the box in the menu.
> 
> can't reproduce this, works well now on amd64 and rpi5

Installed it over the command line.

Worked when installing over the menu => OK

**Review by cstenglein — APPROVED** [review]

**Comment by cstenglein** [comment]
To merge you'll need the approval of a codeowner => @rootzoll

**Pull request merged**